File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/businessmanagementvb/vb/closequote.vb
' <snippetclosequote>


 ' convert the opportunity to a quote
 genQuoteFromOppRequest =
     New GenerateQuoteFromOpportunityRequest With
     {
         .OpportunityId = _opportunityId,
         .ColumnSet = New ColumnSet("quoteid", "name")
     }
 genQuoteFromOppResponse = CType(_serviceProxy.Execute(genQuoteFromOppRequest), 
     GenerateQuoteFromOpportunityResponse)

 Dim closeQuote As Quote = genQuoteFromOppResponse.Entity.ToEntity(Of Quote)()
 _closeQuoteId = closeQuote.Id

 ' Activate the quote
 Dim activateQuote As New SetStateRequest() With
     {
         .EntityMoniker = closeQuote.ToEntityReference(),
         .State = New OptionSetValue(CInt(Fix(QuoteState.Active))),
         .Status = New OptionSetValue(CInt(Fix(quote_statuscode.InProgress)))
     }
 _serviceProxy.Execute(activateQuote)

 ' Close the quote
 Dim closeQuoteRequest_Renamed As New CloseQuoteRequest() With
     {
         .QuoteClose = New QuoteClose() With
                       {
                           .QuoteId = closeQuote.ToEntityReference(),
                           .Subject = "Quote Close " &amp; Date.Now.ToString()
                       },
         .Status = New OptionSetValue(-1)
     }
 _serviceProxy.Execute(closeQuoteRequest_Renamed)

 Console.WriteLine("Quote Closed")

' </snippetclosequote>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/businessmanagement/cs/closequote.cs
// <snippetclosequote>


 // convert the opportunity to a quote
 genQuoteFromOppRequest = new GenerateQuoteFromOpportunityRequest
 {
     OpportunityId = _opportunityId,
     ColumnSet = new ColumnSet("quoteid", "name")
 };
 genQuoteFromOppResponse = (GenerateQuoteFromOpportunityResponse)
     _serviceProxy.Execute(genQuoteFromOppRequest);

 Quote closeQuote = genQuoteFromOppResponse.Entity.ToEntity<Quote>();
 _closeQuoteId = closeQuote.Id;

 // Activate the quote
 SetStateRequest activateQuote = new SetStateRequest()
 {
     EntityMoniker = closeQuote.ToEntityReference(),
     State = new OptionSetValue((int)QuoteState.Active),
     Status = new OptionSetValue((int)quote_statuscode.InProgress)
 };
 _serviceProxy.Execute(activateQuote);

 // Close the quote
 CloseQuoteRequest closeQuoteRequest = new CloseQuoteRequest()
 {
     QuoteClose = new QuoteClose()
     {
         QuoteId = closeQuote.ToEntityReference(),
         Subject = "Quote Close " + DateTime.Now.ToString()
     },
     Status = new OptionSetValue(-1)
 };
 _serviceProxy.Execute(closeQuoteRequest);

 Console.WriteLine("Quote Closed");

// </snippetclosequote>